numbers/monteCarlo.go

```diff
@@ -161,19 +181,21 @@ func RandGamma(a float64, b float64) float64 {
 		e := 1.0 + e1*a
 		for 1 > 0 { //repeat loop until breaks
 			p := e * rand.Float64()
+			rExp, _ = RandExp()
 			if p >= 1.0 {
 				x = -1 * math.Log((e-p)/a)
-				if RandExp() >= (1.0-a)*math.Log(x) {
+				if rExp >= (1.0-a)*math.Log(x) {
 					break
 				}
 			} else {
 				x = math.Exp(math.Log(p) / a)
-				if RandExp() >= x {
+				rExp, _ = RandExp()
```

Comment:
I don't think this line is needed because you generate a number of line 184 that you can use?